src/codecs/cats/crc.c

```diff
@@ -0,0 +1,32 @@
+#include "crc.h"
+
+uint16_t calc_crc(uint8_t *data, size_t length);
+
+size_t cats_append_crc(uint8_t *data, size_t len)
+{
+    uint16_t crc = calc_crc(data, len);
+    data[len++] = crc;
```

Comment:
With this kind of functions we must be careful to stay inside boundaries of allocated memory regions.